include/utils.h

```diff
@@ -3,9 +3,28 @@
 
 #include <err.h>
 #include <stdlib.h>
-
+#include <stdint.h>
+/**
+ * Mark a function as unimplemented and error out when called
+ */
 #define UNIMPLEMENTED()                                                        \
-	errx(EXIT_FAILURE, "function %s in %s:%d not implemented",             \
-	     __FUNCTION__, __FILE__, __LINE__)
+	errx(EXIT_FAILURE, "function %s in %s:%d not implemented", __func__,   \
+	     __FILE__, __LINE__)
+
+/**
+ * Mark argument as `unused`, for example when developing unimplemented functions
+ */
+#define UNUSED(__ARG) __attribute__((__unused__)) __ARG
```

Comment:
why do we need an __ARG ? (     just expand UNUSED -> __attribute__((__unused__))     )